lib/lattice_field.cpp

```diff
@@ -236,32 +240,46 @@ namespace quda {
           qudaDeviceSynchronize();
           comm_barrier();
           for (int b=0; b<2; b++) {
-	    device_pinned_free(ghost_recv_buffer_d[b]);
-	    device_pinned_free(ghost_send_buffer_d[b]);
-	    host_free(ghost_pinned_send_buffer_h[b]);
-	    host_free(ghost_pinned_recv_buffer_h[b]);
-	  }
+#ifndef NVSHMEM_COMMS
```

Comment:
This should moved to targets/cuda/malloc.cpp and in a special allocator `device_comms_pinned_malloc` etc.